maven-scm-api/src/main/java/org/apache/maven/scm/command/changelog/AbstractChangeLogCommand.java

```diff
@@ -46,6 +46,15 @@ protected abstract ChangeLogScmResult executeChangeLogCommand( ScmProviderReposi
                                                                    String datePattern )
         throws ScmException;
 
+    @Deprecated
+    protected ChangeLogScmResult executeChangeLogCommand( ScmProviderRepository repository, ScmFileSet fileSet,
```

Comment:
Is it just the git diff being funky, or are is this PR adding new methods and immediately deprecating them?

Reply:
Correct. One should use the `ChangeLogScmRequest` class.

Reply:
Well, i come from a different angle here. I just want to get the limit parameter to work. Because we have a use case for this. I am not here to refactor this to use ChangeLogScmRequest. So i simply duplicated deprecated methods to make it work. And yes - made them deprecated from the start so that someone doing the refactoring part sees these also have to be taken into account.